### ISSUE TITLE
Generalized Force for spherical joints

### ISSUE THREAD

**Issue by EricVoll** [raisimTech/raisimLib#123]
Hi,

I am trying to use a spherical joint in an ArticulatedSystem as described [here](https://raisim.com/sections/ArticulatedSystem.html).

The state and generalized velocity structure of spherical joints are clear to me (4D quaternions & 3D angular velocity) but the torque/generalized force input via the `setGeneralizedForce` method is not clear to me. Dimension-Wise a spherical joint should have a 3D vector since it is multiplied with the generalized velocity to get the power, but how exactly is it defined? I could not find it in the docs.

In other words: I have a current quaternion and a desired quaternion for a spherical joint. How do I get the generalizedForce for it using a PD controller?

Any help is greatly appreciated. Thanks!

**Comment by jhwangbo** [comment]
The GC should be clear. The GV is the angular velocity expressed in the local frame. This local frame is the one attached to the child of the joint.  As you said, GV dot GF should be power. This means that GF is the torque expressed in the local frame.

Why don't you use the built-in PD controller? PD controller for the spherical joint is complicated to implement and is often unstable

**Comment by EricVoll** [comment]
We are not using the built in PD controller, because we have a Neural Net to model the actuator behavior for AnyMal. The model outputs torque. I am currently using the same Environment but with a different robot for test purposes and would like to keep everything as similar as possible. But in that case I'll switch to the internal PD controller. Thanks for your help!
